frontend/src/algolia-widgets.js

```diff
@@ -0,0 +1,49 @@
+/*!
+ * sri sri guru gaurangau jayatah
+ */
+
+import { connectInfiniteHits } from 'instantsearch.js/es/connectors';
+import Hogan from 'hogan.js';
+import { dom } from './dom-utils';
+import { TempStorage } from './temp-storage';
+
+export class AlgoliaWidgets {
```

Comment:
InstantSearch widgets library uses functional approach, therefore it seems logical to adopt the same style for custom widgets. I'd recommend that we have each widget in a separate file and imported in `app.js` as we import standard widgets from `instantsearch.js/es/widgets`.

But please don't move all existing custom widgets to separate files now. It will be done in another task. However I felt it is necessary to make this note now.

Reply:
i am hoping that you can answer to this one in my previous/other comments.